doc/pac/Port Access Control.md

```diff
@@ -0,0 +1,844 @@
+6. When PAC is disabled on the port, the operationally added untagged VLAN, if present, is removed from the port and the user configured untagged VLAN is assigned back to the port.
+7. If clients are authorized on the port's configured untagged VLAN and the VLAN configuration is modified, all the authenticated clients on the port are removed.
+8. If clients are authorized on RADIUS assigned VLAN, any updates on the port's configured untagged VLAN does not affect the clients. The configuration is updated in the CONFIG_DB but not propagated to the port.
+
+
+### 2.2.6 Warmboot
+
+After a Warm Boot, the authenticated client sessions are torn down and they need to authenticate again.
+
+# 3 Design
+
+## 3.1 Overview
+
+[Figure 2](#configuration-flow) shows the high level design overview of PAC services in SONiC. The existing "macsec" docker is leveraged. 
+
+PAC is composed of multiple sub-modules. 
+
+1. pacd: PAC daemon is the main module that controls client authentication. It is the central repository of PAC clients. It makes use of hostapd and mabd daemons to authenticate clients via 802.1x and MAB respectively.
+
+2. hostapd: This 802.1x module is an opensource Linux application that is available in the SONiC "macsec" docker. It uses hostapd.conf as its config file. 
+
+3. mabd: This is the MAB authentication module.
+
+4. hostapdmgrd: This is the hostapd manager module. It listens to 802.1x specific configurations from CONFIG_DB and translates them to respective hostapd.conf file config entries and commands to hostapd.
+
+
+### 3.1.1 Configuration flow
+
+![pac-config-flow](images/PAC_Config_Flow.JPG)
+
+**Figure 2: PAC service daemon and configuration flow**
+
+1. Mgmt interfaces like CLI write the user provided configuration to CONFIG_DB.
+2. The pacd, mabd and hostapdmgrd gets notified about their respective configurations.
+3. hostapd being a standard Linux application gets its configuration from a hostapd.conf file. hostapdmgrd  generates the hostapd.conf file based on the relevant CONFIG_DB tables. hostapdmgrd informs hostapd about the list of ports it needs to run on. This port list is dynamic as it depends of port link/admin state, port configuration etc. hostapdmgrd  keeps hostapd updated about these changes. 
+4. These modules communicate amongst themselves via socket messages.
+5. hostapd listens to EAPoL PDUs on the provided interface list. When it receives a PDU, it consults pacd and proceeds to authenticate the client. pacd also listens to "unknown src MAC" and triggers MAB, if configured on the port, to authenticate the client.
+
+
+
+### 3.1.2 EAPoL receive flow
+
+![pac-eapol-rx-flow](images/PAC_EAPoL_Rx_Flow.JPG)
+
+
+**Figure 3: EAPoL receive flow**
+
+1. EAPoL packet is received by hardware on a front panel interface and trapped to the CPU by COPP rules for EAP. The packet gets delivered to the hostapd socket listening on EtherType 0x888E.
```

Comment:
How does design handle malicious client attach with filling up with FDB space? I know there is a rate limit 128 discussed however how does it handle in case good clients and many malicious clients (> 129) there is a chance that good client never get a chance to join the system. 
Use case: Lets DoS attach occurred in campus network where all the kids try to connect network while DoS attach occurred,.

Reply:
DoS actually happens as PAC does not attempt authentication beyond 128 clients. It is not an FDB limit that causes the DoS.
PAC needs to have the logic to not even attempt authentication for MACs classified as "malicious". This will be taken up a future release.